eventbroadcaster/notifiersink.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package eventbroadcaster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/eventnotifier"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NotifierSink struct {
+	client   client.Client
+	notifier eventnotifier.Notifier
+}
+
+// RegisterNotifierSinks builds notifiers sinks and registers them on the given broadcaster
+func RegisterNotifierSinks(mgr ctrl.Manager, broadcaster record.EventBroadcaster, notifiersConfig eventnotifier.NotifiersConfig) (err error) {
+	err = nil
+
+	client := mgr.GetClient()
+
+	notifiers, err := eventnotifier.GetNotifiers(notifiersConfig)
+
+	for _, notifier := range notifiers {
+		broadcaster.StartRecordingToSink(&NotifierSink{client: client, notifier: notifier})
+	}
+
+	return
+}
+
+func (s *NotifierSink) Create(event *corev1.Event) (*corev1.Event, error) {
```

Comment:
What happens when this function returns an error? Does it retry or stop here? Because we'll likely reach this case if the username stored in the disruption status is invalid (for instance when a disruption is created by something else than a human, like a workflow).

Reply:
Nothing happens: the error is generated in the slack notifier, gets back to the NotifierSink.Create function, which just absorbs the error (the error from parseEventToNotifier is just ignored, as it would actually generate a "Event Rejected" error log)